### ISSUE TITLE
CS block real-time replication along with proxy get [JIRA: RIAK-1847]

### ISSUE THREAD

**Issue by shino** [basho/riak_repl#320]
Current `proxy_get` configuration defines two behaviors:
1. filters out blocks to be replicated in real-time sync, and
2. enables proxy get blocks from source cluster.

I think real-time block replication even with proxy get functionality may be
useful in some circumstances, such that
- Network bandwidth between data centers is wide enough
- and using Advanced mode (or version 3) replication.

Any comment is appreciated.

---

Implementation sketch:

Add new configuration to `riak_repl`, for example,

```
   {riak_repl, [
         {proxy_get, enabled},
         {cs_block_real_time, enabled}
     ]},
```

and `riak_repl_cs` determines whether replicate blocks in real-time.
(`cs_block_real_time` is unstylish... )

**Comment by bookshelfdave** [comment]
I believe this change would be easy enough to make. I'll add it as a feature request for Riak 2.0.

 The existing block filter is here: https://github.com/basho/riak_repl/blob/master/src/riak_repl_cs.erl

**Comment by cmeiklejohn** [comment]
Moving to 2.1.

**Comment by kuenishi** [comment]
Merged to 2.1 branch at #678 .
